validation: Rename CheckInputs to CheckInputScripts

jnewbery opened the pull request:
CheckInputs() used to check no double spends, scripts & sigs and amounts. Since
832e074, the double spend and amount checks
have been moved to CheckTxInputs(), and CheckInputs() now just validates
input scripts. Rename the function to CheckInputScripts().

Also fix incorrect comments.

jnewbery committed:
scripted-diff: [validation] Rename CheckInputs to CheckInputScripts

CheckInputs() used to check no double spends, scripts & sigs and amounts. Since
832e0744cb8b1e1625cdb19b257f97316ac16a90, the double spend and amount checks
have been moved to CheckTxInputs(), and CheckInputs() now just validates
input scripts. Rename the function to CheckInputScripts().

-BEGIN VERIFY SCRIPT-
sed -i -E -e 's/CheckInputs\b/CheckInputScripts/g' $(git grep -l CheckInputs | grep -v doc/)
-END VERIFY SCRIPT-

jnewbery committed:
[validation] fix comments in CheckInputScripts()

jnewbery commented:
This builds on #13868. Please review that PR first.

fanquake reviewed (commented):
Concept ACK
> This builds on #13868. Please review that PR first.

I have [reviewed](https://github.com/bitcoin/bitcoin/pull/13868#pullrequestreview-282325755) the base PR.

Should 046cd7555e909d9f62a14c44ef0c09eac209b287 be a scripted-diff? This should produce the same change:
```bash
gsed -i 's/\<CheckInputs\>/CheckInputScripts/g' src/*.h src/*.cpp src/**/*.h src/**/*.cpp
```

This needs a couple functional test changes for the expected messages:
https://github.com/bitcoin/bitcoin/blob/7d6f63cc2c2b9c4f07a43619eef0b7314474fffd/test/functional/feature_dersig.py#L127
https://github.com/bitcoin/bitcoin/blob/7d6f63cc2c2b9c4f07a43619eef0b7314474fffd/test/functional/feature_cltv.py#L141

jnewbery commented:
Rebased and removed WIP tag.

Thanks for the review @fanquake 

> This needs a couple functional test changes for the expected messages.

Fixed. Thanks.

> Should 046cd75 be a scripted-diff? 

I think this is small enough to be a non-scripted commit, and I'd need to add more to your suggested one-liner to make the test changes atomic with the validation.cpp changes. I'm leaving as it is for now, but can change if other reviewers agree that scripted would be better.

ajtowns commented:
> I'd need to add more to your suggested one-liner to make the test changes atomic with the validation.cpp changes

FWIW, I believe a valid scripted-diff line for 3623a8f946c55c3fd897a057ac38acb85817f593 would be:

`sed -i -E -e 's/CheckInputs\b/CheckInputScripts/g' $(git grep -l CheckInputs | grep -v doc/)`

MarcoFalke commented on src/validation.cpp:
```diff
@@ -1284,8 +1281,10 @@ void InitScriptExecutionCache() {
+ * Check whether all of this transaction's input scripts succeed.
+ *
+ * This involves ECDSA signature checks so can be computationally intensive. This function should
+ * only be called after the basic tx amount tests in CheckTxInputs passed.
```
This seems to imply that when the basic tx amount tests pass, the amount fields in the tx are correct. However, you can only know that the amount fields are correct when all subsequent check (including signature checks have been run). I'd rather say something like "all cheap sanity checks passed", to be not overly specific.

jnewbery replied:
agree. Fixed

jnewbery commented:
> FWIW, I believe a valid scripted-diff line for 3623a8f would be...

I've updated the commit to use your scripted diff one-liner. Thanks!

michaelfolkson commented:
Concept ACK. I ran a `git grep` on `CheckInputs` in this PR branch and apart from old release notes that obviously won't change, there  is a `ValidateCheckInputsForAllFlags` function in `src/test/txvalidationcache_tests.cpp`, a `CheckInputsAndUpdateCoins` function in `src/txmempool.cpp` and `CheckInputsFromMempoolAndCache` function in `src/validation.cpp`. Should they be changed too?

fanquake commented:
@jnewbery Looks like the scripted diff needs fixing:
```bash
$ set -o errexit; source ./ci/lint/06_script.sh
Error: script block marker but no scripted-diff in title
Failed
```

jnewbery commented:
> Looks like the scripted diff needs fixing

Thanks. Will fix next week.

jnewbery commented:
> Looks like the scripted diff needs fixing

should be fixed

jnewbery commented:
rebased

MarcoFalke commented:
ACK 3ee9ad76803c0f9fdcb21ac1ad6162df79df07da thanks for fixing documentation

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 3ee9ad76803c0f9fdcb21ac1ad6162df79df07da thanks for fixing documentation
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUg1jQwAystQNbOYRs28u4UXVAow/cWruoKgEWgtVxmszf3FIl6XOkkK35LHsXD8
P0YZ/7XAaGIXufQYGkMr4Iaq/lONZjw6oQOqkgCNfq5UjPS4JesEuj/QtKwrQGPp
xFYdGzuU4CE5NQlAtx1bXK5CebOc03FgFYuN9u74579t7nl0P5dp/VjaIUVcdZFE
Wgxgwfa2DboOnI5sv9drQvcX9nLJcUPfBZWufb9RUCE+rklB0hjTomISPUbF75fd
2XFUuOFgIonwwrFkgg6ChTaecDHaphCsttauGRigKxWyOsk3NIjwc8DfAyDK2+H8
HZcMguMNy183bqRFbzbL/VEJD3rDYvj8dtg8cwGlejcP20tj3g1yZqJIerqXU8GZ
5t6xENcNWHjOuL8HpuRg9JiRBM1E12/c39IUAKaG/5W3K4tODI3ziRfu8JQMKpDH
CrDHoaB23mTzzjYxq7R9aMWDSdA0oLzTKL1kGi1Y3SEOWs06e7JSAMhdnoOiueg4
st2HgDxb
=CD9a
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `2fb4b49625388db44b56838d9617fb9d4d07331a02b3a6db6af05658e13fad54  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401082fb4b49625388db44b56838d9617fb9d4d07331a02b3a6db6af05658e13fad54f010fc472a10ee93362d9ff9e179c89e644208fff010379d7e9b92cb300a3f0a8920aad535b208f1045da0d62ef008f19253101679fe230083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff0105ba3337775c8d583fc53a5cebaa809fe08f1045da0d62ef0081357dacde37add590083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010789642e302e9c445f3f1e04c4908a21e08f1045da0d62ff008c467a94430de89230083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010c95b1c8525d2a610e1a9582f4576523a08f1045da0d62ef008c64e27ca02d831a90083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

jnewbery commented:
rebased

MarcoFalke commented:
re-ACK 44d20b86b74445adb6d96f20e44efd1afae36035 only change is rebase the scripted diff

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK 44d20b86b74445adb6d96f20e44efd1afae36035 only change is rebase the scripted diff
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUhQBgv/f7OOsZwi+esjzjBssh0Bf6JMXRjjbZ+ARDo7KEqKjlf6x09B1ViMqC+X
+emq0IRC9dNzDLW6foblXUGvNwwsBCoULWmI7gXL+m6wZxkD2Byiuq1RDrLLvty1
cadiGJUbhIrMzUHXPsYPx/nt/Wq3kRc9UX09DU/JbkjBcZYYYsrMy96mqPt+2q+e
/hQ/1JsfofAqqS6cFNSNvnxSUG/vywJb6dgKd/c5+2SYAuZ1Nl4eTQiTeueP67nL
lypZt55WRlWoGo+1zUQIE6dDQxkfyNJfdsRQa8tNWS8wRdsBJNafuNRPiRpTcV6T
PJKq/J5syT7AYBQiyfyLyqYKaPve4j/nR4opIBtnZZzc02eQZH0MnHPhpamzV+LO
/Mqm5IYPC7xR5csWzTJas7lRaum81DWSKVowL7MO69M+TUmKH2K4n7vZ0kj1YKCY
dxvTZIzkglBR7uD9vRKUPv7KhVTNPiCSm1SreLyUSJJuE82Mex1JXAXt4Hj7UUSH
foyxWkSG
=oHWs
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `0630b39dafbd879abe6610f1a999b11bd28a1e6e9f1c97ebdfa437aa875e56fa  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401080630b39dafbd879abe6610f1a999b11bd28a1e6e9f1c97ebdfa437aa875e56faf010e81558c9269fc430663fae18348c639c08fff01000859c5d804f4292ab58155b019dc4f108f120319c513b08a5d86f3270e75f4d1b7b768de219b23aa615aa856ea387a0013d0308f1045db31861f00874d8ec1dbf8638160083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0100ed0506494420e7a19c5d4e226ee901d08f1045db31861f008f38d1ba299ef45030083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010a31f0a5749bfce2b51276fcb1ea42a2e08f1045db31861f00843f611faefaea4490083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010f32d134fd2a04582044adfed5f3a44ad08f1045db31862f0086690db5d6841bea90083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

promag commented:
TIL `git grep -l CheckInputs -- ':!doc'`.

Code review ACK.

jnewbery commented:
Rebased on master

MarcoFalke commented:
This is a trivial fix that should be ready for merge after rebase

jnewbery commented:
Rebased

MarcoFalke commented:
re-ACK 3bd8db80d8d335ab63ece4f110b0fadd562e80b7, did the rebase myself, checked the scripted diff 👡

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

re-ACK 3bd8db80d8d335ab63ece4f110b0fadd562e80b7, did the rebase myself, checked the scripted diff 👡
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUjh4wv/WSY4eMrYXocYlXm6kXbisKRQM+4z5HS9drtint6Mf3u/Zj3zioAxEApI
sIk6tuwzracrSYy2D6YtZnSSjpMwrtCRYy48OOt8e0WDnlUpZwTUyjCwp+l4nsMa
Bc+MUtTe9Eq8ICEAyFbNVGr+yvt18WlqM/h5DEIchFlGIE76BajPBTRxnf1QVe5o
OiC7ErXnpj79LzBYP5MF8/R7AMb66y+++Xop9pZeMkLAWM//0exTlsGrJU7V/GJW
2Mx/wApRUFjTz/NNDl5iArpayOa7KFcPZ9MhzvfBV0v+vihH3HzDKmvT9NYTDgox
6WvsBr1nj2F1ONV/fZQ9h8SK0jj4ewguDgNJ8cwB2uJQBqUOaWayNNNbE1egJIOv
33VpHuFdUIHsuRJMufZPpuNrK2/h0HguHVunuIM8jIvdG0dt+ioFYabXmKkywPi8
EQjw/j4DyvGzKILVKFpHatAFHLET87oGMH2Ph7ComQKOmYjsuyXvnys04y6KOwlg
4n7UJzLs
=3RbB
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `4aa6710c657c2553e2de927e5918efcd786d8c666d5ddcdb193f75684907f609  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401084aa6710c657c2553e2de927e5918efcd786d8c666d5ddcdb193f75684907f609f010084ece3e66c402266654c7dd90a24cd208fff01002b27a47b04e01c7c3d2c5296c2ab58f08f1045dc46aadf00847bc65062d7d8ecd0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010515fa8cbad5d053c1806fe670c586a3c08f1045dc46aadf00876ca0ba3da3d5f510083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6dfff0106120f37c0f5e9482f53a7336e6e14adb08f1045dc46aadf0080043861eec3395920083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010d35dbc57200c84a216e6722a861bcfc108f1045dc46aadf00854450d49baadb0b80083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

promag commented:
ACK 3bd8db80d8d335ab63ece4f110b0fadd562e80b7 :trollface:
